### PR TITLE
Added non-const mutable accessors for world child objects

### PR DESCRIPTION
**Pull request by nkoenig**
Signed-off-by: Nate Koenig <nate@openrobotics.org>

# 🎉 New feature

## Summary

Similar to #838. This adds non-const accessors to the World class.

## Test it
Run the tests

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Updated documentation (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by azeey — COMMENTED** [review]
Similar to my comment in the other PR, did you leave out mutable `World::Frame` intentionally?

Otherwise, LGTM!

**Comment by nkoenig** [comment]
Added Frame modifiers in [4affb5e](https://github.com/ignitionrobotics/sdformat/pull/840/commits/4affb5e9b44702c7ad08e9f0b206f6bc32597d4e)

**Review by azeey — APPROVED** [review]
LGTM! Just a couple minor comments

File: src/World_TEST.cc [hunk 1]
```diff
@@ -376,6 +377,39 @@ TEST(DOMWorld, AddModel)
...
+
+  sdf::Frame *mutableFrame = world.FrameByIndex(0);
+  mutableFrame->SetName("newName1");
+  EXPECT_EQ(mutableFrame->Name(), world.FrameByIndex(0)->Name());
+
+  sdf::Frame *mutableFrameByName = world.FrameByName("frame1");
+  EXPECT_EQ(nullptr, mutableFrameByName);
+  mutableFrameByName = world.FrameByName("newName1");
+  ASSERT_NE(nullptr, mutableFrameByName);
+  EXPECT_EQ(mutableFrameByName->Name(), world.FrameByName("newName1")->Name());
```
[the remark this answers is not in the document]
Comment (nkoenig) [reply src/World_TEST.cc]: Done in [5aa06b6](https://github.com/ignitionrobotics/sdformat/pull/840/commits/5aa06b63effc102f68084893976db017d4250508)

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-04-13-fortress-edifice/1367/1

**Pull request merged**